crates/ruff_python_parser/src/string.rs

```diff
@@ -110,99 +130,101 @@ impl<'a> StringParser<'a> {
-                        self.get_pos(),
-                    ))
-                }
-            }
-        }
 
-        if name.len() > MAX_UNICODE_NAME {
+        let Some('{') = self.next_char() else {
+            return Err(LexicalError::new(LexicalErrorType::StringError, start_pos));
+        };
+
+        let start_pos = self.get_pos();
+        let Some(close_idx) = memchr::memchr(b'}', self.rest.as_bytes()) else {
             return Err(LexicalError::new(
-                LexicalErrorType::UnicodeError,
+                LexicalErrorType::StringError,
                 self.get_pos(),
             ));
-        }
+        };
+
+        let name_and_ending = self.skip_bytes(close_idx + 1);
+        let name = &name_and_ending[..name_and_ending.len() - 1];
 
-        unicode_names2::character(&name)
+        unicode_names2::character(name)
             .ok_or_else(|| LexicalError::new(LexicalErrorType::UnicodeError, start_pos))
     }
 
-    fn parse_escaped_char(&mut self) -> Result<String, LexicalError> {
-        match self.next_char() {
-            Some(c) => {
-                let char = match c {
-                    '\\' => '\\',
-                    '\'' => '\'',
-                    '\"' => '"',
-                    'a' => '\x07',
-                    'b' => '\x08',
-                    'f' => '\x0c',
-                    'n' => '\n',
-                    'r' => '\r',
-                    't' => '\t',
-                    'v' => '\x0b',
-                    o @ '0'..='7' => self.parse_octet(o),
-                    'x' => self.parse_unicode_literal(2)?,
-                    'u' if !self.kind.is_any_bytes() => self.parse_unicode_literal(4)?,
-                    'U' if !self.kind.is_any_bytes() => self.parse_unicode_literal(8)?,
-                    'N' if !self.kind.is_any_bytes() => self.parse_unicode_name()?,
-                    // Special cases where the escape sequence is not a single character
-                    '\n' => return Ok(String::new()),
-                    '\r' => {
-                        if self.peek() == Some('\n') {
-                            self.next_char();
-                        }
-                        return Ok(String::new());
-                    }
-                    c => {
-                        if self.kind.is_any_bytes() && !c.is_ascii() {
-                            return Err(LexicalError {
-                                error: LexicalErrorType::OtherError(
-                                    "bytes can only contain ASCII literal characters".to_owned(),
-                                ),
-                                location: self.get_pos(),
-                            });
-                        }
-                        return Ok(format!("\\{c}"));
-                    }
-                };
-                Ok(char.to_string())
-            }
-            None => Err(LexicalError {
+    fn parse_escaped_char(&mut self, string: &mut String) -> Result<(), LexicalError> {
```

Comment:
For another PR: It would be nice if we wouldn't need to allocate a `String` when no character gets replaced (e.g. `\n` remains unchanged).

Reply:
Yeah, agree. I planned to implement such a feature in a subsequent PR by trying to use `Cow<_>` and considering other methods (e.g. `smol_str`/`compact_str`).

Reply:
Cow feels nice but is a bit painful because it also requires adding lifetimes to all AST nodes. I'm not against it, just be prepared to deal with 1000's of lifetime issues

Reply:
Yeah, the diff would be an eyesore. However, I am wary of the possible lack of change in performance from using an inline string crate. I think we do enough reading of the strings in the linting/formatting stage to make the control flow required enough to mess up our performance from my previous attempt to introduce stack strings.